gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/retention/HivePartitionVersionRetentionCleaner.java

```diff
@@ -0,0 +1,129 @@
+ * limitations under the License.
+ */
+package gobblin.compliance.retention;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.compliance.ComplianceConfigurationKeys;
+import gobblin.compliance.HiveProxyQueryExecutor;
+import gobblin.compliance.purger.HivePurgerQueryTemplate;
+import gobblin.compliance.utils.PartitionUtils;
+import gobblin.compliance.utils.ProxyUtils;
+import gobblin.configuration.State;
+import gobblin.data.management.retention.dataset.CleanableDataset;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.util.HadoopUtils;
+
+
+/**
+ * A version cleaner for the {@link HivePartitionRetentionVersion}.
+ *
+ * A version will be considered as clean only if it's metadata no longer exists in the db and the data
+ * it was pointing to no longer exists.
+ *
+ * @author adsharma
+ */
+@Slf4j
+public class HivePartitionVersionRetentionCleaner extends HivePartitionVersionRetentionRunner {
+  private FileSystem fs;
+  private boolean simulate;
+  private Optional<String> versionOwner = Optional.absent();
+
+  public HivePartitionVersionRetentionCleaner(CleanableDataset dataset, DatasetVersion version,
+      List<String> nonDeletableVersionLocations, State state) {
+    super(dataset, version, nonDeletableVersionLocations, state);
+    this.versionOwner = ((HivePartitionRetentionVersion) this.datasetVersion).getOwner();
+    this.simulate = this.state.getPropAsBoolean(ComplianceConfigurationKeys.COMPLIANCE_JOB_SIMULATE,
+        ComplianceConfigurationKeys.DEFAULT_COMPLIANCE_JOB_SIMULATE);
+  }
+
+  /**
+   * If simulate is set to true, this will simply return.
+   * If version is pointing to an empty location, drop the partition and close the jdbc connection.
+   * If version is pointing to the same location as of the dataset, then drop the partition and close the jdbc connection.
+   * If version is pointing to the non deletable version locations, then drop the partition and close the jdbc connection.
+   * Otherwise delete the data underneath, drop the partition and close the jdbc connection.
+   */
+  @Override
+  public void clean()
+      throws IOException {
+    Path versionLocation = ((HivePartitionRetentionVersion) this.datasetVersion).getLocation();
+    Path datasetLocation = ((CleanableHivePartitionDataset) this.cleanableDataset).getLocation();
+    String completeName = ((HivePartitionRetentionVersion) this.datasetVersion).datasetURN();
+
+    State state = new State(this.state);
+    this.fs = ProxyUtils.getOwnerFs(state, this.versionOwner);
+
+    try (HiveProxyQueryExecutor queryExecutor = ProxyUtils.getQueryExecutor(state, this.versionOwner)) {
+      log.info("Trying to clean version " + completeName);
+      if (this.simulate) {
```

Comment:
If putting the `simulate` check here, the following logic that checks if this version should be deleted would be skipped, and we cannot tell if this version will be deleted if not running `simulate` mode.

Instead, if you put `simulate` check around line 98 and 100, which skips the real `delete` and `drop` actions when `simulate` is enabled, we can have more accurate logs when running `simulate` mode.